packages/features/ee/video/ViewRecordingsDialog.tsx

```diff
@@ -0,0 +1,159 @@
+import { useSession } from "next-auth/react";
+import { useState } from "react";
+
+import dayjs from "@calcom/dayjs";
+import LicenseRequired from "@calcom/features/ee/common/components/v2/LicenseRequired";
+import { WEBSITE_URL } from "@calcom/lib/constants";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { RouterOutputs, trpc } from "@calcom/trpc/react";
+import type { PartialReference } from "@calcom/types/EventManager";
+import type { RecordingObj } from "@calcom/types/VideoApiAdapter";
+import { Dialog, DialogClose, DialogContent, DialogFooter, DialogHeader } from "@calcom/ui";
+import { Loader, Button, showToast, Icon } from "@calcom/ui";
+
+import TeamsUpgradeRecordingBanner from "./components/TeamUgradeRecordingBanner";
+
+type BookingItem = RouterOutputs["viewer"]["bookings"]["get"]["bookings"][number];
+
+interface IViewRecordingsDialog {
+  booking?: BookingItem;
+  isOpenDialog: boolean;
+  setIsOpenDialog: React.Dispatch<React.SetStateAction<boolean>>;
+  timeFormat: number | null;
+}
+
+function convertStoMs(seconds: number) {
+  // Bitwise Double Not is faster than Math.floor
+  const minutes = ~~(seconds / 60);
+  const extraSeconds = seconds % 60;
+  return `${minutes}min  ${extraSeconds}sec`;
+}
+
+interface GetTimeSpanProps {
+  startTime: string | undefined;
+  endTime: string | undefined;
+  locale: string;
+  hour12: boolean;
+}
+
+const getTimeSpan = ({ startTime, endTime, locale, hour12 }: GetTimeSpanProps) => {
+  if (!startTime || !endTime) return "";
+  return (
+    new Intl.DateTimeFormat(locale, { hour: "numeric", minute: "numeric", hour12 }).format(
+      new Date(startTime)
+    ) +
+    " - " +
+    new Intl.DateTimeFormat(locale, { hour: "numeric", minute: "numeric", hour12 }).format(new Date(endTime))
+  );
+};
+
+export const ViewRecordingsDialog = (props: IViewRecordingsDialog) => {
+  const { t, i18n } = useLocale();
+  const { isOpenDialog, setIsOpenDialog, booking, timeFormat } = props;
+  const [downloadingRecordingId, setRecordingId] = useState<string>("");
```

Comment:
```suggestion
  const [downloadingRecordingId, setRecordingId] = useState<string | null>(null);
```

Reply:
This is optional but we can handle our initial state with null to represent that it hasn't being initialised.